client/state/sites/actions.js

```diff
@@ -158,6 +159,13 @@ export function requestSite( siteFragment, forceWpcom = false ) {
 				) {
 					return dispatch( requestSite( siteFragment, true ) );
 				}
+				if ( returnError ) {
+					return dispatch( {
+						type: SITE_REQUEST_FAILURE,
+						errorCode: error.error,
+						statusCode: error.status,
+					} );
+				}
```

Comment:
It would be much better if `requestSite` returned a reasonable promise. I.e., resolve with site info on success, and reject with the error on failure. That would require checking all usages of `dispatch( requestSite( siteId ) ).then( ... )` and making them work with rejected promises, too.

Reply:
I think that'd make sense, but I feel it might be a little outside the scope of this PR, especially since it's only really dealing with failures in a specific scenario and there's a risk of causing issues elsewhere by changing that.

Reply:
> I feel it might be a little outside the scope of this PR

That's true, it's complex enough to deserve its own PR. I just created #53187 for that.

Reply:
@Aurorum I merged #53187 today and I see you already merged `trunk` 🙂 This PR still does a trivial whitespace change in `client/state/sites/actions.js`. Would be nice to clean that up.

Reply:
Testing this again, and I think there's a couple more tweaks needed to make this work with #53187 too. I'll try and fix those and then fix up the whitespace at the same time. :)

Reply:
@jsnajdr Should all be working now - please let me know if there's any remaining issues! Thanks for working on #53187 too! :)